packages/agreement-process/src/app.ts

```diff
@@ -18,8 +22,10 @@ app.disable("x-powered-by");
 
 app.use(healthRouter);
 app.use(contextMiddleware(serviceName));
+app.use(await applicationAuditBeginMiddleware(serviceName, config));
 app.use(authenticationMiddleware(config));
 app.use(loggerMiddleware(serviceName));
+app.use(await applicationAuditEndMiddleware(serviceName, config));
```

Comment:
Minor: Since the endMiddleware uses `res.on("finish")`, there is no need to put it after other middleware: I would group the two to have them one near the other in the chain (here and everywhere else)

```suggestion
app.use(await applicationAuditBeginMiddleware(serviceName, config));
app.use(await applicationAuditEndMiddleware(serviceName, config));
app.use(authenticationMiddleware(config));
app.use(loggerMiddleware(serviceName));
```